### PR TITLE
Minor code cleanup in GifImagePlugin

### PR DESCRIPTION
**Pull request by radarhere**
Inspired by @homm's comments in #2196

### CODE REVIEW

File: PIL/GifImagePlugin.py [hunk 1]
```diff
@@ -439,13 +439,9 @@ def _get_local_header(fp, im, offset, flags):
...
-                for i in range(len(used_palette_colors)):
-                    if used_palette_colors[i] == transparency:
-                        transparency = i
-                        transparent_color_exists = True
-                        break
-                    else:
-                        transparent_color_exists = False
+                transparent_color_exists = transparency in used_palette_colors
+                if transparent_color_exists:
+                    transparency = used_palette_colors.index(transparency)
```
Comment (hugovk) [inline PIL/GifImagePlugin.py]: This refactored line isn't covered by tests:
https://travis-ci.org/python-pillow/Pillow/jobs/182957414#L4912

Is it possible to test it?

File: PIL/GifImagePlugin.py [hunk 2]
```diff
@@ -439,13 +439,9 @@ def _get_local_header(fp, im, offset, flags):
...
             if len(used_palette_colors) < 256:
-                for i in range(len(used_palette_colors)):
-                    if used_palette_colors[i] == transparency:
-                        transparency = i
-                        transparent_color_exists = True
-                        break
-                    else:
-                        transparent_color_exists = False
+                transparent_color_exists = transparency in used_palette_colors
+                if transparent_color_exists:
```
Comment (hugovk) [inline PIL/GifImagePlugin.py]: `if transparency in used_palette_colors:`

Comment (radarhere) [inline PIL/GifImagePlugin.py]: You're suggesting that I combine the two lines, removing the variable? If so, it's used in lines 450-452.

Comment (hugovk) [inline PIL/GifImagePlugin.py]: Ah yes, I missed that. Ignore this then :)

File: PIL/GifImagePlugin.py [hunk 3]
```diff
@@ -439,13 +439,9 @@ def _get_local_header(fp, im, offset, flags):
...
             # adjust the transparency index after optimize
             if len(used_palette_colors) < 256:
-                for i in range(len(used_palette_colors)):
-                    if used_palette_colors[i] == transparency:
-                        transparency = i
-                        transparent_color_exists = True
-                        break
-                    else:
-                        transparent_color_exists = False
+                transparent_color_exists = transparency in used_palette_colors
```
Comment (homm) [inline PIL/GifImagePlugin.py]: It's not a big deal, but there are two array sequence scans: first in this line, second in `used_palette_colors.index`. Is it possible to do `used_palette_colors.index` first and use its result?

Comment (radarhere) [inline PIL/GifImagePlugin.py]: Okay, I've updated the PR.

File: PIL/GifImagePlugin.py (hunk 1 above)
Comment (radarhere) [inline PIL/GifImagePlugin.py]: I have updated the PR, correcting a difference in flow. From what I can see, the lines are now covered by tests.

**Comment by wiredfool** [comment]
closed in favor of #2374

**Pull request closed without merging**